src/server/zset_family.cc

```diff
@@ -1058,6 +1058,38 @@ struct SetOpArgs {
   bool with_scores = false;
 };
 
+void HandleOpStatus(ConnectionContext* cntx, OpStatus op_status) {
+  switch (op_status) {
+    case OpStatus::INVALID_FLOAT:
+      return (*cntx)->SendError("weight value is not a float", kSyntaxErrType);
+    default:
+      return (*cntx)->SendError(op_status);
+  }
+}
+
+OpResult<ScoredMap> IntersectResults(const vector<OpResult<ScoredMap>>& results, AggType agg_type) {
+  OpResult<ScoredMap> result;
+  for (auto op_res : results) {
+    if (op_res.status() == OpStatus::SKIPPED)
+      continue;
+
+    if (!op_res) {
+      return op_res.status();
+    }
+
+    if (op_res->empty()) {
+      return op_res.status();
+    }
+
+    if (result.value().empty()) {
```

Comment:
I think that there's a bug in this implementation...
Think of the case where we're intersecting the following:
(1) `[a, b, c]`
(2) `[d, e, f]`
(3) `[f]`

We'll get an empty intersection between (1) and (2), but because none of them is empty we'll continue, and this `if` will swap an empty result with `[f]`, but that's a bug.

We'll need to come up with another way to initialize `result`, maybe make it `optional<>` and then if it's not yet set, do the swap? Or add some `bool init = false;` to track that?

Reply:
Could you also add a test for that case please?
Make sure that it fails before you fix it, and passes afterwards

Reply:
Good catch @chakaz . I see that the original code had:
```
if (result.empty())
      break;
```
at the end of the loop.

Reply:
Thanks for the review!
I added the test. However, the test did not fail before the fix.
Currently investigating the reason.
https://github.com/dragonflydb/dragonfly/pull/2164/commits/8e9919026de701d62b5d21dc9b76b4ad851a4a6e

Reply:
Thank you for bearing with me/us :)
Please let me know when you're ready for another review, or if you'd like some help with the failing test

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/00842a80fcc12b92febd0c2330b4a08882c56722#diff-294d1289b1741ae808c06f38d260077a53dcba7762c845f9677a068568821fc9R1080-R1091